ignite/pkg/cosmosgen/generate_typescript.go

```diff
@@ -0,0 +1,228 @@
+package cosmosgen
+
+import (
+	"github.com/ignite-hq/cli/ignite/pkg/cosmosanalysis/module"
+	"github.com/ignite-hq/cli/ignite/pkg/giturl"
+	"github.com/ignite-hq/cli/ignite/pkg/gomodulepath"
+	"github.com/ignite-hq/cli/ignite/pkg/nodetime/programs/sta"
+	tsproto "github.com/ignite-hq/cli/ignite/pkg/nodetime/programs/ts-proto"
+	"github.com/ignite-hq/cli/ignite/pkg/protoc"
+
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	tsOut = []string{
+		"--ts_proto_out=.",
+	}
+
+	jsOpenAPIOut = []string{
+		"--openapiv2_out=logtostderr=true,allow_merge=true,json_names_for_fields=false,Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:.",
+	}
+)
+
+type tsGenerator struct {
+	g *generator
+}
+
+type generatePayload struct {
+	Modules []module.Module
+	User    string ``
+	Repo    string ``
+}
+
+func newTSGenerator(g *generator) *tsGenerator {
+	return &tsGenerator{
+		g: g,
+	}
+}
+
+func (g *generator) generateTS() error {
+	tsg := newTSGenerator(g)
+
+	chainPath, _, err := gomodulepath.Find(g.appPath)
+	if err != nil {
+		return err
+	}
+
+	chainInfo, err := giturl.Parse(chainPath.RawPath)
+	if err != nil {
+		return err
+	}
+
+	data := generatePayload{
+		Modules: g.appModules,
+		User:    chainInfo.User,
+		Repo:    chainInfo.Repo,
+	}
+
+	if g.o.jsIncludeThirdParty {
+		for _, modules := range g.thirdModules {
+			data.Modules = append(data.Modules, modules...)
+		}
+	}
+
+	if err := tsg.generateModuleTemplates(); err != nil {
+		return err
+	}
+
+	if err := tsg.generateVueTemplates(data); err != nil {
+		return err
+	}
+
+	if err := tsg.generateRootTemplates(data); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (g *tsGenerator) generateModuleTemplates() error {
+	tsprotoPluginPath, cleanup, err := tsproto.BinaryPath()
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	gg := &errgroup.Group{}
+
+	generate := func(sourcePath string, modules []module.Module) {
```

Comment:
Why is the loop wrapped with `generate` function. It seems it could be simplified.